### PR TITLE
Add Notification model and handle notification event

### PR DESCRIPTION
**Pull request by raman325**
I plan to submit a PR similar to https://github.com/home-assistant/core/pull/45729 where we fire an event when a notification is received for a node.

### CODE REVIEW

File: zwave_js_server/model/node.py [hunk 1]
```diff
@@ -366,6 +367,9 @@ def handle_metadata_updated(self, event: Event) -> None:
 
     def handle_notification(self, event: Event) -> None:
         """Process a node notification event."""
+        event.data["notification"] = Notification(
```
Comment (MartinHjelmare) [inline zwave_js_server/model/node.py:370]: I think we should change the key above when storing the `ValueNotification` to `"value_notification"` to be consistent.

Comment (raman325) [reply zwave_js_server/model/node.py:370]: @marcelveldt ^ guess I shouldn't have crossed out my comment on the other PR 🙂 

Do we do that as part of this PR or a separate one?

Comment (MartinHjelmare) [reply zwave_js_server/model/node.py:370]: I think it's ok to do it here. We can bump and release after this is merged.

**Comment by MartinHjelmare** [comment]
Please rebase to solve the merge conflict.

File: zwave_js_server/model/node.py (hunk 1 above)
Comment (marcelveldt) [reply zwave_js_server/model/node.py:370]: Go ahead and change here. I'll adjust my PR

Comment (marcelveldt) [reply zwave_js_server/model/node.py:370]: BTW: I more and more vote for changing the models with prefixes such as ZwaveNode and ZwaveValue and NodeNotification and ValueNotification, see #25

**Comment by marcelveldt** [comment]
@raman325 maybe use the same logic to fire the event ? I can adjust my PR so it will also forward these events ?

**Comment by raman325** [comment]
The event data will be different but otherwise yeah that makes sense. I was thinking it would be nice eventually to tie these events to the primary "entity" but that might be easier said than done.

File: zwave_js_server/model/node.py (hunk 1 above)
Comment (MartinHjelmare) [reply zwave_js_server/model/node.py:377]: Yes, we can do that clean up after this PR if merged.

**Comment by MartinHjelmare** [comment]
Entities don't fire events, besides the built in state changed events, in Home Assistant.

**Review by MartinHjelmare — APPROVED** [review]
Are we happy to merge?

**Comment by raman325** [comment]
sure I meant add entity_id as part of the event data to use as criteria for automations

**Comment by MartinHjelmare** [comment]
We can discuss that.

**Pull request merged**